templates/taxi/car_detail.html

```diff
@@ -1,22 +1,30 @@
 {% extends "base.html" %}
 
 {% block content %}
-  <h1>
-    {{ car.model }}
-    <a href="{% url 'taxi:car-delete' pk=car.id %}" class="btn btn-danger link-to-page">
-      Delete
-    </a>
-
-    <a href="{% url 'taxi:car-update' pk=car.id %}" class="btn btn-secondary link-to-page">
-      Update
-    </a>
-  </h1>
-  <p>Manufacturer: ({{ car.manufacturer.name }}, {{ car.manufacturer.country }})</p>
-  <h1>Drivers</h1>
-  <hr>
+  <h2>{{ car.model }} - Details</h2>
+  <p>Manufacturer: {{ car.manufacturer }}</p>
+  
+  <h3>Assigned Drivers:</h3>
   <ul>
     {% for driver in car.drivers.all %}
       <li>{{ driver.username }} ({{ driver.first_name }} {{ driver.last_name }})</li>
+    {% empty %}
+      <li>No drivers assigned</li>
     {% endfor %}
   </ul>
+
+  <!-- Check if the current user is already a driver for this car -->
+  {% if request.user in car.drivers.all %}
+    <form method="post" action="{% url 'taxi:assign-remove-driver' car.pk %}">
+      {% csrf_token %}
+      <button type="submit" class="btn btn-danger">Remove me from this car</button>
```

Comment:
you use here backward relationship but did you fetched it from db? because otherwise here will be N+1 issue